### PR TITLE
Cargo Vaurca

### PR DESCRIPTION
**Pull request by Geevies**
* Blank Vaurca Drones ordered from cargo now start with an active filter port, meaning they won't immediately start suffocating.
* Blank Vaurca Drones now start with the correct hive colour.
* C'thur is now one of the possible randomly generated hive names.

Note to maintainers: Blank Vaurca Drones' Item Data will have to be changed.

New description:
`A brain dead, generic vaurca drone.`
New path:
`/mob/living/carbon/human/type_a/cargo`

It's not vital, but at the moment, the access requirement for these is `47`, which is Science access. Sometimes, medical orders these to teach newbies how to do surgery on vaurca. Thus it would be nice to have it be `47` OR `5`, which is medical access. If my cargo bodybag PR goes through, this access restriction should be removed entirely.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Geevies** [comment]
(Testing done prior to adding the C'thur as a possible hive)
image

File: code/modules/mob/living/carbon/human/species/station/vaurca/vaurca_subspecies.dm [hunk 1]
```diff
@@ -145,7 +145,7 @@
 		)
 
 	has_organ = list(
-		"neural socket"       = /obj/item/organ/vaurca/neuralsocket,
+		BP_NEURAL_SOCKET       = /obj/item/organ/vaurca/neuralsocket,
 		BP_LUNGS              = /obj/item/organ/internal/lungs/vaurca,
 		BP_HEART              = /obj/item/organ/internal/heart/vaurca,
 		"phoron reservoir"    = /obj/item/organ/vaurca/reservoir,
```
Comment (mikomyazaki) [inline code/modules/mob/living/carbon/human/species/station/vaurca/vaurca_subspecies.dm:151]: Why not do the reservoir too while we're doing defines?

File: code/modules/mob/language/station.dm [hunk 2]
```diff
@@ -176,7 +176,7 @@
 	var/new_name = "[pick(list("Ka'","Za'","Ka'"))]"
 	new_name += "[pick(list("Akaix'","Viax'"))]"
 	new_name += "[pick(list("Uyek","Uyit","Avek","Theth","Ztak","Teth","Zir","Yek","Zirk","Ayek","Yir","Kig","Yol","'Zrk","Nazgr","Yet","Nak","Kiihr","Gruz","Guurz","Nagr","Zkk","Zohd","Norc","Agraz","Yizgr","Yinzr","Nuurg","Iii","Lix","Nhagh","Xir","Z'zit","Zhul","Zgr","Na'k","Isk'yet","Aaaa"))]"
-	new_name += " [pick(list("Zo'ra","Zo'ra","Zo'ra","K'lax"))]"
+	new_name += " [pick(list("Zo'ra","Zo'ra","Zo'ra","K'lax","C'thur"))]"
```
Comment (mikomyazaki) [inline code/modules/mob/language/station.dm:179]: This should be a weighted list.

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

**Comment by Geevies** [comment]
image

**Review by mikomyazaki — APPROVED** [review]

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 3]
```diff
@@ -94,15 +94,15 @@
 			H.adjustToxLoss(-2 * removed)
 			return
 
-		if(alien == IS_VAURCA && H.species.has_organ["filtration bit"])
+		if(alien == IS_VAURCA && H.species.has_organ[BP_NEURAL_SOCKET])
 			metabolism = REM * 20 //vaurcae metabolise phoron faster than other species - good for them if their filter isn't broken.
-			var/obj/item/organ/vaurca/filtrationbit/F = H.internal_organs_by_name["filtration bit"]
+			var/obj/item/organ/vaurca/filtrationbit/F = H.internal_organs_by_name[BP_NEURAL_SOCKET]
```
Comment (Arrow768) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:99]: filtration bit vs neural socket

**Review by Arrow768 — CHANGES_REQUESTED** [review]

File: code/modules/organs/subtypes/vaurca.dm [hunk 4]
```diff
@@ -56,22 +56,22 @@
...
+	organ_tag = BP_PHORON_RESERVOIR
 	parent_organ = BP_CHEST
 	icon_state = "phoron_reservoir"
 	robotic = 1
 
 /obj/item/organ/vaurca/filtrationbit
-	name = "filtration bit"
-	organ_tag = "filtration bit"
+	name = BP_NEURAL_SOCKET
+	organ_tag = BP_NEURAL_SOCKET
```
Comment (Arrow768) [inline code/modules/organs/subtypes/vaurca.dm:67]: filtrationbit vs neural socket

File: code/modules/mob/living/carbon/human/species/station/vaurca/vaurca_subspecies.dm [hunk 5]
```diff
@@ -145,16 +145,16 @@
...
-		"mechanical liver"    = /obj/item/organ/internal/liver/vaurca/robo,
-		"mechanical kidneys"  = /obj/item/organ/internal/kidneys/vaurca/robo,
+		BP_PHORON_RESERVOIR    = /obj/item/organ/vaurca/reservoir,
+		BP_VAURCA_LIVER    = /obj/item/organ/internal/liver/vaurca/robo,
+		BP_VAURCA_KIDNEYS  = /obj/item/organ/internal/kidneys/vaurca/robo,
 		BP_STOMACH            = /obj/item/organ/internal/stomach,
 		BP_BRAIN              = /obj/item/organ/internal/brain/vaurca,
 		BP_EYES               = /obj/item/organ/internal/eyes/vaurca,
-		"filtration bit"      = /obj/item/organ/vaurca/filtrationbit
+		BP_NEURAL_SOCKET      = /obj/item/organ/vaurca/filtrationbit
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/species/station/vaurca/vaurca_subspecies.dm:157]: filtrationbit vs neural socket

File: code/modules/mob/living/carbon/human/species/station/vaurca/vaurca.dm [hunk 6]
```diff
@@ -93,11 +93,11 @@
 	stamina_recovery = 2	//slow recovery
 
 	has_organ = list(
-		"neural socket"        = /obj/item/organ/vaurca/neuralsocket,
+		BP_NEURAL_SOCKET        = /obj/item/organ/vaurca/neuralsocket,
 		BP_LUNGS               = /obj/item/organ/internal/lungs/vaurca,
-		"filtration bit"       = /obj/item/organ/vaurca/filtrationbit,
+		BP_NEURAL_SOCKET       = /obj/item/organ/vaurca/filtrationbit,
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/species/station/vaurca/vaurca.dm:98]: filtration bit vs neural socket

File: code/modules/mob/living/carbon/human/life.dm [hunk 7]
```diff
@@ -629,7 +629,7 @@
 
 		var/total_phoronloss = 0
 		for(var/obj/item/I in src)
-			if(I.contaminated && !(isvaurca(src) && src.species.has_organ["filtration bit"]))
+			if(I.contaminated && !(isvaurca(src) && src.species.has_organ[BP_NEURAL_SOCKET]))
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/life.dm:632]: filtration bit vs neural socket

**Review by Arrow768 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human_species.dm [hunk 8]
```diff
@@ -65,11 +65,28 @@ INITIALIZE_IMMEDIATE(/mob/living/carbon/human/dummy/mannequin)
 	. = ..(mapload, SPECIES_MONKEY_VAURCA)
 	src.gender = FEMALE
 
-/mob/living/carbon/human/type_a/Initialize(mapload)
+/mob/living/carbon/human/type_a/Initialize(mapload, var/new_species = null)
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/human_species.dm:68]: Whats the purpose of that extra argument here (that is then never used)?

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**